cvmfs/telemetry_aggregator_influx.cc

```diff
@@ -0,0 +1,245 @@
+                                    influx_metric_name_.c_str(),
+                                    influx_extra_tags_.c_str(),
+                                    influx_extra_fields_.c_str());
+  } else {
+    is_zombie_ = true;
+    LogCvmfs(kLogTelemetry, kLogDebug | kLogSyslogWarn,
+             "Not enabling influx metrics. Not all mandatory variables set: "
+             "CVMFS_INFLUX_METRIC_NAME, CVMFS_INFLUX_HOST, CVMFS_INFLUX_PORT");
+  }
+}
+
+/**
+ * Influx dataformat
+ * ( https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/ )
+ * Syntax
+   <measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>]
+ *
+ * Example
+   myMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000
+*/
+std::string TelemetryAggregatorInflux::MakePayload() {
+  // measurement and tags
+  std::string ret = "" + influx_metric_name_ + "_absolute,repo=" + fqrn_;
+
+  if (influx_extra_tags_ != "") {
+    ret += "," + influx_extra_tags_;
+  }
+
+  // fields
+  ret += " ";
+  const int64_t revision = counters_["catalog_revision"];
+  ret += "catalog_revision=" + StringifyUint(revision);
+
+  std::string tok = ",";
+  for (std::map<std::string, int64_t>::iterator it
+      = counters_.begin(), iEnd = counters_.end(); it != iEnd; it++) {
+    if (it->second != 0) {
+      ret += tok + it->first + "=" + StringifyInt(it->second);
+
+      tok = ",";
+    }
+  }
+  if (influx_extra_fields_ != "") {
+    ret += tok + influx_extra_fields_;
+  }
+
+  // timestamp
+  ret += " " + StringifyUint(timestamp_);
+
+  return ret;
+}
+
+/**
+ * Influx dataformat
+ * ( https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/ )
+ * Syntax
+   <measurement>[,<tag_key>=<tag_value>[,<tag_key>=<tag_value>]] <field_key>=<field_value>[,<field_key>=<field_value>] [<timestamp>]
+ *
+ * Example
+   myMeasurement,tag1=value1,tag2=value2 fieldKey="fieldValue" 1556813561098000000
+*/
+std::string TelemetryAggregatorInflux::MakeDeltaPayload() {
+  // measurement and tags
+  std::string ret = "" + influx_metric_name_ + "_delta,repo=" + fqrn_;
+
+  if (influx_extra_tags_ != "") {
+    ret += "," + influx_extra_tags_;
+  }
+
+  // fields
+  ret += " ";
+  const uint64_t revision = counters_["catalog_revision"];
+  ret += "catalog_revision=" + StringifyUint(revision);
+
+  std::string tok = ",";
+  for (std::map<std::string, int64_t>::iterator it
+      = counters_.begin(), iEnd = counters_.end(); it != iEnd; it++) {
+    int64_t value = it->second;
+    int64_t old_value = old_counters_.at(it->first);
+    if (value != 0) {
+      ret += tok + it->first + "=" + StringifyInt(value - old_value);
+    }
+  }
+
+  if (influx_extra_fields_ != "") {
+    ret += tok + influx_extra_fields_;
+  }
+
+  // timestamp
+  ret += " " + StringifyUint(timestamp_);
+
+  return ret;
+}
+
+TelemetryError TelemetryAggregatorInflux::SendToInflux(
```

Comment:
I think that we should determine the host address and open the socket only once.